addons/project/models/project.py

```diff
@@ -2062,6 +2062,9 @@ def write(self, vals):
             self.check_access_rule('write')
             portal_can_write = True
 
+        if 'project_id' in vals:
+            self.env['project.project'].browse(vals['project_id']).check_access_rule('write')
```

Comment:
Hello 👋🏼 

I am not sure to understand why we have to check the user can write on the project when we change the project of a task. 🤔 
We could check if he can read the project to set on that task but I am not sure it is really useful. 🤔 

May I miss something? 🧐

Reply:
Okay, I see, however, I think we should check if the user can read the project to set instead of checking if he can write on it.

Moreover, if the user cannot access to that project, he cannot really select it (in the many2one field) in the UI, right? If yes, then I don't think it is really needed. 🙂 

What do you think? 🙂